### PR TITLE
[structural_application] commit new structural_application to compile with pybind and amatrix

### PR DESCRIPTION
**Pull request by vryy**

### CODE REVIEW

File: applications/structural_application/custom_python/add_custom_processes_to_python.cpp [hunk 1]
```diff
@@ -49,16 +48,17 @@ namespace Kratos
...
-    class_<TopologyUpdateProcess, bases<Process>, boost::noncopyable >
-    ("TopologyUpdateProcess", init<ModelPart&, double, double, double>())
+    py::class_<TopologyUpdateProcess, Process >
+    (m, "TopologyUpdateProcess")
+    .def(py::init<ModelPart&, double, double, double>())
     .def("SetBinSize", &TopologyUpdateProcess::SetBinSize)
     .def("GetTopologyChange", &TopologyUpdateProcess::GetTopologyChange)
     .def("GetObjective", &TopologyUpdateProcess::GetObjective)
-    .def(self_ns::str(self))
+    // .def(self_ns::str(self))
```
Comment (philbucher) [inline applications/structural_application/custom_python/add_custom_processes_to_python.cpp]: here you could use `PrintObject` from `includes/define_python.h`

**Review by philbucher — COMMENTED** [review]
- python exposure seems ok
- changes are local only to the structural-app
- no objections from my side, approval I leave to @RiccardoRossi

**Review by RiccardoRossi — APPROVED** [review]
thank you for the time and effort

**Comment by vryy** [comment]
As a post comment, is it possible to automatically trigger the test server to include the new application? It seems that it compiled again just the applications defined in sample configure script.

**Comment by philbucher** [comment]
which apps are compiled there is decided by the @KratosMultiphysics/technical-committee 
however currently it is not possible to add new apps bcs the CI is already at it's (time) limits, even now we often have the problem that the builds get cancelled because they take too long
Furthermore I think that an app has to fulfill certain requirements in order to be added, but from what I know a more detailed explanation regarding this will come after the workshop end of the month.

**Comment by RiccardoRossi** [comment]
This is just to confirm that we are at the limit of available time, so that as of now we cannot add new stuff to the CI.

We are in the process of adding our own machines for CI so to be able to relax these requirements. We will speak about this at the Kratos Workshop

**Pull request merged**